cleanlab_studio/utils/tlm_calibrated.py

```diff
@@ -0,0 +1,173 @@
+        Use `Studio.TLMCalibrated()` instead of this method to initialize a TLMCalibrated object.
+        lazydocs: ignore
+        """
+        self._api_key = api_key
+
+        if quality_preset not in {"base", "low", "medium"}:
+            raise ValidationError(
+                f"Invalid quality preset: {quality_preset}. TLMCalibrated only supports 'base', 'low' and 'medium' presets."
+            )
+        self._quality_preset = quality_preset
+
+        self._options = options
+        self._timeout = timeout if timeout is not None and timeout > 0 else None
+        self._verbose = verbose
+
+        custom_eval_criteria_list = (
+            self._options.get("custom_eval_criteria", []) if self._options else []
+        )
+
+        # number of custom eval critera + 1 to account for the default TLM trustworthiness score
+        self._num_features = len(custom_eval_criteria_list) + 1
+        self._rf_model = RandomForestRegressor(monotonic_cst=[1] * self._num_features)
+
+        self._tlm = TLM(
+            self._api_key,
+            quality_preset=self._quality_preset,
+            options=self._options,
+            timeout=self._timeout,
+            verbose=self._verbose,
+        )
+
+    def fit(self, tlm_scores: List[TLMScore], ratings: Sequence[float]) -> None:
+        """
+        Callibrate the model using TLM scores obtained from a previous `TLM.get_trustworthiness_score()` call
+        using the provided numeric ratings.
+
+        Args:
+            tlm_scores (List[TLMScore]): list of [TLMScore](../trustworthy_language_model/#class-tlmscore) object obtained
+                from a previous `TLM.get_trustworthiness_score()` call
+            ratings (Sequence[float]): sequence of numeric ratings corresponding to each prompt-response pair,
+                the length of this sequence must match the length of the `tlm_scores`.
+        """
+        if len(tlm_scores) != len(ratings):
+            raise ValidationError(
+                "The list of ratings must be of the same length as the list of TLM scores."
+            )
+
+        tlm_scores_df = pd.DataFrame(tlm_scores)
+        extracted_scores = self._extract_tlm_scores(tlm_scores_df)
+
+        if extracted_scores.shape[1] != self._num_features:
+            raise ValidationError(
+                f"TLMCalibrated has {self._num_features - 1} custom evaluation criteria defined, "
+                f"however the tlm_scores provided have {extracted_scores.shape[1] - 1} custom evaluation scores. "
+                "Please make sure the number of custom evaluation criterias match."
+            )
+
+        # using pandas so that NaN values are handled correctly
+        ratings_series = pd.Series(ratings)
+        ratings_normalized = (ratings_series - ratings_series.min()) / (
+            ratings_series.max() - ratings_series.min()
+        )
+
+        self._rf_model.fit(extracted_scores, ratings_normalized.values)
+
+    def get_trustworthiness_score(
```

Comment:
could be in followup PRs since i believe we want to release this asap, but I think we should also support the `prompt` function for this new class

Reply:
created https://www.notion.so/cleanlab/support-prompt-and-try_-functions-for-TLMCalibrated-10cc7fee85be8070b2c9ff018df04d18